src/mbodied_agents/agents/motor/rt1/rt1_agent.py

```diff
@@ -0,0 +1,225 @@
+from mbodied_agents.agents.motor.rt1.transformer_network import TransformerNetwork
+from mbodied_agents.base.motion import Motion
+from mbodied_agents.types.controls import HandControl, JointControl, Pose
+from mbodied_agents.types.vision import SupportsImage
+from transformers import BertModel, BertTokenizer
+
+observation_space = spaces.Dict(
+    {
+        "image_primary": spaces.Box(low=0.0, high=1.0, shape=(3, 224, 224), dtype=np.float32),
+        "natural_language_embedding": spaces.Box(low=-np.inf, high=np.inf, shape=[768], dtype=np.float32),
+    },
+)
+action_space_dict = OrderedDict(
+    [
+        (
+            "xyz",
+            spaces.Box(low=-1, high=1, shape=(3,), dtype=np.float32),
+        ),
+        (
+            "rpy",
+            spaces.Box(low=-np.pi, high=np.pi, shape=(3,), dtype=np.float32),
+        ),
+        (
+            "grasp",
+            spaces.Box(low=0, high=1.0, shape=(1,), dtype=np.float32),
+        ),
+    ],
+)
+
+
+class RT1Agent(MotorAgent):
+    """RT1Agent class responsible for interacting with the environment based on the agent's policy.
+
+    This agent uses a TransformerNetwork to generate actions based on the given observations
+    and context (natural language embedding).
+
+    Attributes:
+        config (dict): Configuration dictionary containing parameters for the agent.
+        device (torch.device): The device to run the computations (CPU or CUDA).
+        model (TransformerNetwork): The neural network model used for predicting actions.
+        policy_state (Optional[torch.Tensor]): The internal state of the policy network.
+        action_tokenizer (RT1ActionTokenizer): The action tokenizer for converting output to actions.
+        image_history (List[torch.Tensor]): History of the past observations.
+        step_num (int): Keeps track of the number of steps taken by the agent.
+    """
+
+    def __init__(self, config, weights_path: str = None, **kwargs) -> None:
+        """Initializes the RT1Agent with the provided configuration and model weights.
+
+        Args:
+            config (dict): Configuration parameters for setting up the agent.
+            weights_path (str, optional): Path to the pre-trained model weights. Defaults to None.
+            **kwargs: Additional keyword arguments.
+
+        Example:
+            config = {
+                "observation_history_size": 6,
+                "future_prediction": 6,
+                "token_embedding_dim": 512,
+                "causal_attention": True,
+                "num_layers": 6,
+                "layer_size": 512,
+            }
+            agent = RT1Agent(config, weights_path="path/to/weights.pth")
+        """
+        self.config = config
+        self.device = torch.device(
+            "cuda" if torch.cuda.is_available() else "cpu"
+        )
+        model_name = 'bert-base-uncased'
+        self.tokenizer = BertTokenizer.from_pretrained(model_name)
```

Comment:
dont reinitialize every call

Reply:
I have this in the constructor / `__init__` method. So its only initialized once when we create the `Rt1Agent` object